src/main/java/org/apache/ibatis/datasource/pooled/PooledDataSource.java

```diff
@@ -581,7 +581,6 @@ protected boolean pingConnection(PooledConnection conn) {
         if (!realConn.getAutoCommit()) {
           realConn.rollback();
         }
-        result = true;
```

Comment:
Removal of true here will result in it being false thus changing the result.  Please add that one back.  I think the other removal looks ok as every path otherwise sets a value.

Reply:
`if (result && poolPingEnabled && poolPingConnectionsNotUsedFor >= 0
        && conn.getTimeElapsedSinceLastUse() > poolPingConnectionsNotUsedFor)`

Look at line 571, it's true.

BTW  this fix is an automatic detection prompt from IDEA.